sellable/vehicle.go

```diff
@@ -0,0 +1,177 @@
+package sellable
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/wopta/goworkspace/lib"
+	"github.com/wopta/goworkspace/models"
+	prd "github.com/wopta/goworkspace/product"
+)
+
+// Function for exposing the sellability function.
+// It takes a request in which the body contains the policy in JSON format,
+// and then it returns the appropriate product.
+func VehicleHandler(w http.ResponseWriter, r *http.Request) (string, interface{}, error) {
+	var (
+		policy models.Policy
+		err    error
+	)
+
+	log.Println("Vehicle")
+
+	bytes, err := io.ReadAll(r.Body)
+	if err != nil {
+		return "{}", nil, fmt.Errorf("cannot read body: %v", err)
+	}
+	if err = json.Unmarshal(bytes, &policy); err != nil {
+		return "{}", nil, err
+	}
+
+	authToken, err := models.GetAuthTokenFromIdToken(r.Header.Get("Authorization"))
+	lib.CheckError(err)
+	product, productJson, err := Vehicle(authToken.Role, policy)
+
+	return productJson, product, err
+}
+
+// Given a policy that shoudl contain the Vehicle asset, then it returns:
+//   - the product or parts of it depending on the sellable rules
+//   - its representation in json
+//   - and an eventual error
+func Vehicle(role string, p models.Policy) (models.Product, string, error) {
```

Comment:
the function should return a product and an error. The marshaling of the product and its output as a string should be done on the handler function.
This will be the function used by the quote/gap endpoint as to not make an unnecessary http call between functions.